packages/calypso-e2e/src/hooks/recording.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { unlink, renameSync } from 'fs';
+import { Context } from 'mocha';
+
+/**
+ * Internal dependencies
+ */
+import { getTargetLocale, getTargetScreenSize } from '../browser-manager';
+import { getDateString, getVideoDir } from '../media-helper';
+
+export async function clearFailedTest( this: Context ): Promise< void > {
+	// String here would be more logical as combination of magellan + mocha should result
+	// in at most one failure per run of mocha.
+	// However, using a string here causes the recorded failures to not survive the context
+	// when the after hook is run whereas an array would.
+	this.failedTest = [];
+}
+
+/**
+ * Tracks failed tests per suite.
+ *
+ * @param {Context} this Current mocha context at the test level.
+ * @returns {void} No return value.
+ */
+export async function recordFailedTestName( this: Context ): Promise< void > {
+	// Guard against the attribute not existing.
+	if ( ! this.currentTest || ! this.currentTest.state || ! this.currentTest.parent ) {
+		return;
+	}
+
+	const state: string = this.currentTest.state.toUpperCase();
+	if ( state === 'FAILED' ) {
+		const fullName =
+			this.currentTest?.parent.title.toString() + ' ' + this.currentTest?.title.toString();
+		this.currentTest.parent.ctx.failedTest.push( fullName );
+	}
+}
+
+/**
+ * Hook to handle video recording.
+ *
+ * If all tests in the suite passes, the video recording is deleted.
+ * If any tests in the suite fails, the video recording is saved.
+ * The video recording will cover the suite from beginning to end.
+ *
+ * @param {Context} this Current mocha context at the suite level.
+ * @returns {void} No return value.
+ */
+export async function saveVideo( this: Context ): Promise< void > {
+	// Guard against page object not being present in this context.
+	if ( ! this.page ) {
+		return;
+	}
+
+	// Obtain the default save path that Playwright generates for the page.
+	const defaultVideoPath: string = await this.page.video().path();
+
+	// Close the BrowserContext so the video is written to disk.
```

Comment:
I'd suggest moving this to its own hook, and here thrown an error if the `BrowserContext` has not been closed yet.

Reply:
I thought about this for a bit and I see why you suggest this - say browser context crashes, or is closed prematurely, then hook actions can fail or behave unpredictably. Is that the risk of simply leaving the code as is?

Reply:
My main concern was that nothing in `saveVideo` suggests the context will be closed. A developer may add another hook after `saveVideo` expecting the context to exist.

Reply:
Ah got it, thanks for clarifying! I will try to restructure this.